src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/BeneficialOwnerChangeService.java

```diff
@@ -150,10 +149,20 @@ private BeneficialOwnerChange<CorporateBeneficialOwnerPsc> covertBeneficialOwner
     Pair<PscApi, PrivateBoDataApi> publicPrivateBoPair = publicPrivateBo.get(
         beneficialOwnerCorporateDto.getChipsReference());
 
-    if (publicPrivateBoPair == null || publicPrivateBoPair.getRight() == null) {
-      ApiLogger.errorContext(SERVICE, NO_PSC_FOUND, null);
+    if (publicPrivateBoPair == null) {
+      ApiLogger.errorContext(SERVICE, NO_PUBLIC_AND_NO_PRIVATE_BO_DATA_FOUND, null, logMap);
       return null;
     }
+    else {
+      if (publicPrivateBoPair.getLeft() == null) {
+        ApiLogger.errorContext(SERVICE, NO_PUBLIC_BO_DATA_FOUND, null, logMap);
```

Comment:
Maybe this case (and similar) below could say "continuing" somehow? As we don't return null;